packages/frontend/src/features/post/hooks/useBackgroundReputationClaim/useBackgroundReputationClaim.ts

```diff
@@ -0,0 +1,196 @@
+import { UnirepSocialCircuit } from '@unirep-app/circuits/dist/src/types'
+import { ReportHistory } from '@/features/reporting/utils/types'
+import { UserState } from '@unirep/core'
+
+type ClaimReputationParams = {
+    report: ReportHistory
+    repUserType: RepUserType
+}
+
+const REFETCH_INTERVAL = 60000
+
+export function useBackgroundReputationClaim() {
+    const queryClient = useQueryClient()
+    const { getGuaranteedUserState } = useUserState()
+
+    const { data: reportsWaitingForTransaction, refetch: refetchReports } =
+        useQuery({
+            queryKey: [QueryKeys.ReportsWaitingForTransaction],
+            queryFn: fetchReportsWaitingForTransaction,
+            refetchInterval: REFETCH_INTERVAL,
+        })
+
+    const generateProof = useCallback(
+        async (
+            userState: UserState,
+            report: ReportHistory,
+            repUserType: RepUserType,
+        ) => {
+            const epochKeyProof = await userState.genEpochKeyProof()
+            const currentEpoch = epochKeyProof.epoch
+            const currentNonce = epochKeyProof.nonce
+            const attesterId = epochKeyProof.attesterId
+            const chainId = userState.chainId
+            const identitySecret = userState.id.secret
+
+            if (repUserType === RepUserType.VOTER) {
+                const reportNullifier = genNullifier(
+                    userState.id,
+                    Number(report.reportId),
+                )
+                const reportNullifierCircuitInputs =
+                    genReportNullifierCircuitInput({
+                        reportNullifier,
+                        identitySecret,
+                        reportId: 1,
+                        currentEpoch,
+                        currentNonce,
+                        attesterId,
+                        chainId,
+                    })
+                return genProofAndVerify(
+                    UnirepSocialCircuit.reportNullifierProof,
+                    reportNullifierCircuitInputs,
+                )
+            } else {
+                const reporterEpochKey =
+                    repUserType === RepUserType.REPORTER
+                        ? report.reportorEpochKey
+                        : report.respondentEpochKey
+                const reportNonNullifierCircuitInput =
+                    genReportNonNullifierCircuitInput({
+                        reportedEpochKey: reporterEpochKey,
+                        identitySecret,
+                        reportedEpoch: 0,
+                        currentEpoch,
+                        currentNonce,
+                        chainId,
+                        attesterId,
+                    })
+                return genProofAndVerify(
+                    UnirepSocialCircuit.reportNonNullifierProof,
+                    reportNonNullifierCircuitInput,
+                )
+            }
+        },
+        [],
+    )
```

Comment:
I suggest to implement helpers called`genNullifierProof` and `genNonNullifierProof` separately. Refer to https://github.com/social-tw/social-tw-website/blob/main/packages/frontend/src/features/core/utils/genReportIdentityProof.ts